test/rules/always_put_control_body_on_new_line.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N always_put_control_body_on_new_line`
+
+testIfElse() {
```

Comment:
No, the `{` can be on the same line as the control flow keyword and the `}` can be on the same line as the `{`.

I have to say, though, I'm not sure that a linter is the right place for this kind of thing. I think the main value of a linter is in finding semantic problems with the code. Syntactic issues are much more driven by aesthetic and it's much harder to apply hard-and-fast rules. The style guide's overriding concern is that the code be readable. If there's some case where having a return on the same line as an if is clearly a win, I wouldn't want to have to then follow it with an `//ignore`, since that would defeat the entire purpose.